infrastructure/cdn-in-a-box/optional/vnc/Dockerfile

```diff
@@ -14,15 +14,22 @@
 # KIND, either express or implied.  See the License for the
 # specific language governing permissions and limitations
 # under the License.
-FROM docker.io/centos:7
+
+ARG CENTOS_VERSION=8
+FROM centos:${CENTOS_VERSION}
+ARG CENTOS_VERSION=8
```

Comment:
duplicate ARG

Reply:
It's not a duplicate `ARG`. `ARG` is the only Dockerfile instruction that is allowed before `FROM`, and in order to pull the image `centos:${CENTOS_VERSION}` where `CENTOS_VERSION` is a build arg, `ARG` needs to go before `FROM`.

However, build args (unlike environment variables) are not accessible in build stages after the one where the build arg is declared. So, in order to be able to use `CENTOS_VERSION` *after* the `FROM` instruction, a second `ARG` instruction is required for `CENTOS_VERSION`.

Reply:
fair enough, but there are multi-stage builds where I think it's doing something like:
```dockerfile
ARG CENTOS_VERSION=8
FROM centos:${CENTOS_VERSION}
ARG CENTOS_VERSION=8
# ...
ENV CENTOS_VERSION=${CENTOS_VERSION}
```
could you instead do
```dockerfile
ARG CENTOS_VERSION=8
ENV CENTOS_VERSION=${CENTOS_VERSION}
FROM centos:${CENTOS_VERSION}
```
?

Reply:
Nope because `ENV` cannot come before the first `FROM`. If it does, you'll get an error like `No build stage in current context`.

From [the documentation](https://docs.docker.com/engine/reference/builder/#from):
> - `ARG` is the only instruction that may precede `FROM` in the `Dockerfile`.
  See [Understand how ARG and FROM interact](https://docs.docker.com/engine/reference/builder/#understand-how-arg-and-from-interact).

Reply:
right, right, you just said that. Alright, then nevermind.